### PR TITLE
.NET Core to .NET branding changes

### LINKED ISSUE #33680: Breaking change: Update FrameworkName to ".NET"
See: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.cs#L11

This [code](https://github.com/richlander/testapps/blob/master/versioninfo/Program.cs) produces this output today (for 5.0):

**.NET Core info**
Version: 5.0.0
FrameworkDescription: .NET Core 5.0.0-preview.1.20120.5
CoreCLR Build: 5.0.0-preview.1.20120.5
CoreCLR Hash: 3c523a6a7a3ebc25fe524359127b1d8846e23ea3
CoreFX Build: 5.0.0-preview.1.20120.5
CoreFX Hash: 3c523a6a7a3ebc25fe524359127b1d8846e23ea3

### PR DESCRIPTION
**Pull request by jkotas**
Contributes to #33680

### CODE REVIEW

**Comment by jkotas** [comment]
This renames about 1/3 of all .NET Core references in the repo to just .NET. I have made sure that all user visible strings are renamed. The remaining instances are in docs or comments, and renaming them to just .NET would often be confusing, e.g. when they refer to specific version or difference against .NET Framework.

**Comment by jkotas** [comment]
Marking as breaking change. This can break code that checks for ".NET Core" in framework description, etc.

**Comment by jkotas** [comment]
Depends on https://github.com/dotnet/arcade/pull/5086

File: docs/design/features/assemblyloadcontext.md
Comment (pergardebrink) [inline docs/design/features/assemblyloadcontext.md]: Was the `Every .NET Core app` supposed to be Core?

File: docs/project/copyright.md
Comment (stephentoub) [inline docs/project/copyright.md]: Is this correct? (It may be, but I want to double-check.)

File: docs/design/features/assemblyloadcontext.md
Comment (joaofx) [inline docs/design/features/assemblyloadcontext.md]: Should it be "Every .NET app has a..."?

File: docs/project/dogfooding.md
Comment (ericstj) [inline docs/project/dogfooding.md]: This is actually verbatim command line output from the CLI. Can we update it to match the rebranded CLI output so versions make sense?  Otherwise hold off on this portion of update.

File: docs/project/dogfooding.md
Comment (ericstj) [inline docs/project/dogfooding.md]: We should update this to current release.  /cc @viktorhofer

File: docs/project/dogfooding.md
Comment (ericstj) [inline docs/project/dogfooding.md]: Should update

File: src/installer/pkg/projects/netcoreapp/sfx/bundle.wxl
Comment (ericstj) [inline src/installer/pkg/projects/netcoreapp/sfx/bundle.wxl]: Is the target of this link updated?

File: src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json
Comment (stephentoub) [inline src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json]: The email is for `dotnetpackages@dotnetfoundation.org` and the name is `.NET Team`.  Should the name be `.NET Foundation` or something like that?

File: README.md
Comment (steveharter) [inline README.md]: What are the rules on why "runtime" in ".NET runtime" is sometimes uppercase and sometimes lowercase?

File: docs/design/features/localization-options.md
Comment (daniel-white) [inline docs/design/features/localization-options.md]: Shouldn’t this one have a replaced noun?

File: docs/design/features/net-core-applications.md
Comment (daniel-white) [inline docs/design/features/net-core-applications.md]: I’d clarify something about the .net framework not being supported

File: docs/project/dogfooding.md
Comment (daniel-white) [inline docs/project/dogfooding.md]: Should update

File: docs/project/linux-performance-tracing.md
Comment (daniel-white) [inline docs/project/linux-performance-tracing.md]: Perhaps do 5.0.0

File: docs/project/linux-performance-tracing.md
Comment (daniel-white) [inline docs/project/linux-performance-tracing.md]: Should update version

**Comment by jkotas** [comment]
Blocked by https://github.com/dotnet/BenchmarkDotNet/pull/1399

File: docs/project/copyright.md
Comment (richlander) [reply docs/project/copyright.md]: Great question ...

I would do two things:

- Link to the actual [license](https://github.com/dotnet/runtime/blob/master/LICENSE.TXT) to make the whole thing concrete.
- Use the similar phrasing as is used in the following project ... "The .NET project ..."

File: src/installer/pkg/projects/netcoreapp/sfx/bundle.wxl
Comment (jkotas) [reply src/installer/pkg/projects/netcoreapp/sfx/bundle.wxl]: It is not yet. I expect that we will update the naming in the docs once we are close to the release. We do not have a system in place that allows us to update the docs for unreleased stuff without causing confusion.

File: README.md
Comment (jkotas) [reply README.md]: I do not think there is any rule. It depends on context in number of cases, e.g. ".NET Runtime and libraries" would look a bit odd in this specific case.

File: src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json
Comment (jkotas) [reply src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json]: This was introduced in https://github.com/dotnet/core-setup/pull/2525/ . @richlander Is there anything to fix?

File: docs/design/features/net-core-applications.md
Comment (jkotas) [reply docs/design/features/net-core-applications.md]: I am deleting this whole file. It is not up to date, and it covers topics that are covered in other places with more details.

File: docs/project/linux-performance-tracing.md
Comment (jkotas) [reply docs/project/linux-performance-tracing.md]: This tracing documentation is spread over multiple places. I believe it would be best to delete this doc, and centralize the documentation on the official docs site. I have opened https://github.com/dotnet/diagnostics/issues/933 on it.

File: src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json
Comment (richlander) [reply src/installer/pkg/packaging/deb/dotnet-runtime-deps-debian_config.json]: ".NET Team" looks good to me as the "maintainer".

**Review by richlander — APPROVED** [review]

**Pull request merged**
